modules/cosmosBidAdapter.js

```diff
@@ -0,0 +1,398 @@
+import { registerBidder } from '../src/adapters/bidderFactory';
+import { BANNER, VIDEO } from '../src/mediaTypes';
+import * as utils from '../src/utils';
+
+const BIDDER_CODE = 'cosmos';
+const BID_ENDPOINT = '//bid.cosmoshq.com/openrtb2/bids';
+const USER_SYNC_ENDPOINT = '//sync.cosmoshq.com/js/v1/usersync.html';
+const HTTP_POST = 'POST';
+const LOG_PREFIX = 'COSMOS: ';
+const DEFAULT_CURRENCY = 'USD';
+const HTTPS = 'https:';
+const MEDIA_TYPES = 'mediaTypes';
+const DEFAULT_NET_REVENUE = false;
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, VIDEO],
+
+  /**
+   * generate UUID
+   **/
+  _createUUID: function () {
+    var u = Date.now().toString(16) + Math.random().toString(16) + '0'.repeat(16);
+    return [u.substr(0, 8), u.substr(8, 4), '4000-8' + u.substr(13, 3), u.substr(16, 12)].join('-');
+  },
+
+  /**
+   * copy object if not null
+   **/
+  _copyObject: function (src, dst) {
+    if (src) {
+      // copy complete object
+      Object.keys(src).forEach(param => dst[param] = src[param]);
+    }
+  },
+
+  /**
+   * parse object
+   **/
+  _parse: function (rawPayload) {
+    try {
+      if (rawPayload) {
+        return JSON.parse(rawPayload);
+      }
+    } catch (ex) {
+      utils.logError(LOG_PREFIX, 'Exception: ', ex);
+    }
+    return null;
+  },
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   **/
+  isBidRequestValid: function (bid) {
+    var retval = false;
+    do {
+      if (!bid || !bid.params) {
+        utils.logError(LOG_PREFIX, 'nil/empty bid object');
+        break;
+      }
+
+      if (!utils.isEmpty(bid.params.publisherId) ||
+        !utils.isNumber(bid.params.publisherId)) {
+        utils.logError(LOG_PREFIX, 'publisherId is mandatory and must be numeric. Ad Unit: ', JSON.stringify(bid));
+        break;
+      }
+      // video bid request validation
+      if (bid.params.hasOwnProperty('video')) {
+        if (!bid.params.video.hasOwnProperty('mimes') ||
+          !utils.isArray(bid.params.video.mimes) ||
```

Comment:
Is this the correct property to check for the `mimes` property?  

From looking at the sample video adUnit in the corresponding md file, the `mimes` property is listed under the `mediaTypes.video` property, not under the `bids[].params.video` property.  If the md file is incorrect and this field should be checked at this location, can you please update the md file?

Reply:
Thanks for good catch. corrected all the checks.